### PR TITLE
Implement `r_indicator`

### LINKED ISSUE #305: [FEATURE] Fix and test `r_indicator` implementation
**Location:** `balance/stats_and_plots/weighted_comparisons_stats.py` (TODO at L42)
https://github.com/facebookresearch/balance/blob/6d34f0120b8531a67aab4bcc1c8f38dbe15cbf79/balance/stats_and_plots/weighted_comparisons_stats.py#L42

The TODO notes that `r_indicator` is broken. This function is used in diagnostics and affects representativeness metrics.

Investigate the current `r_indicator` logic, identify the failure mode, and implement a corrected version. Add unit tests covering typical use cases and edge cases (e.g., identical samples, extreme weight distributions).

### PR DESCRIPTION
**Pull request by neuralsorcerer**
- Closes #305

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,15 @@
-# 0.18.0 (2026-03-24)
+# 0.18.0 (Unreleased - TBD)
```
[the remark this answers is not in the document]
Comment (neuralsorcerer) [reply CHANGELOG.md]: Updated :)
